File: ObjectiveC/FlexChart101/FlexChart101/ChartData.m
//
//  ChartData.m
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

#import "ChartData.h"

@implementation ChartData

- (id)initWithName:(NSString *)name sales:(NSNumber *)sales expenses:(NSNumber *)expenses downloads:(NSNumber *)downloads {
    self = [super init];
    if(self){
        _name = name;
        _sales = sales;
        _expenses = expenses;
        _downloads = downloads;
    }
    return self;
}   
+ (NSNumber *)generateRandom:(NSInteger) max {
    return [NSNumber numberWithUnsignedInteger:(arc4random() % max)];
}

+ (NSMutableArray *)demoData {
    ChartData *US = [[ChartData alloc] initWithName:@"US" sales:[ChartData generateRandom : 10001] expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Germany = [[ChartData alloc] initWithName:@"Germany" sales:[ChartData generateRandom : 10001] expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *UK = [[ChartData alloc] initWithName:@"UK" sales:[ChartData generateRandom : 10001] expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Japan = [[ChartData alloc] initWithName:@"Japan" sales:[ChartData generateRandom : 10001] expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Italy = [[ChartData alloc] initWithName:@"Italy" sales:[ChartData generateRandom : 10001] expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Greece = [[ChartData alloc] initWithName:@"Greece" sales:[ChartData generateRandom : 10001] expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    return [[NSMutableArray alloc] initWithObjects:US, Germany, UK, Japan, Italy, Greece, nil];
}

+ (NSMutableArray *)annotationData {
    ChartData *US = [[ChartData alloc] initWithName:@"US" sales:@1500 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Germany = [[ChartData alloc] initWithName:@"Germany" sales:@4000 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *UK = [[ChartData alloc] initWithName:@"UK" sales:@3000 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Japan = [[ChartData alloc] initWithName:@"Japan" sales:@6000 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Italy = [[ChartData alloc] initWithName:@"Italy" sales:@3500 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Greece = [[ChartData alloc] initWithName:@"Greece" sales:@8500 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *France = [[ChartData alloc] initWithName:@"France" sales:@2300 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Spain = [[ChartData alloc] initWithName:@"Spain" sales:@6500 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Ireland = [[ChartData alloc] initWithName:@"Ireland" sales:@4500 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    ChartData *Poland = [[ChartData alloc] initWithName:@"Poland" sales:@9500 expenses:[ChartData generateRandom : 5001] downloads:[ChartData generateRandom : 20001]];
    
    return [[NSMutableArray alloc] initWithObjects:US, Germany, UK, Japan, Italy, Greece, France, Spain, Ireland, Poland, nil];
}

@end

@implementation ChartPoint

- (id)initWithX:(NSNumber*)x y:(NSNumber*)y {
    self = [super init];
    if(self){
        _x = x;
        _y = y;
    }
    return self;
}

+ (NSMutableArray *)generateRandomPoints:(NSInteger)count {
    NSMutableArray *points = [[NSMutableArray alloc] init];
    
    if (count % 2 == 1) {
        count++;
    }

    for (int i = 0; i < count / 2; i++) {
        do
        {
            double random = rand() / (double)RAND_MAX;
            double u = 2 * random - 1;
            double v = 2 * random - 1;
            
            double s = u * u + v * v;
            
            if (s < 1)
            {
                s = sqrt(-2 * log(s) / s);
                [points addObject:[[ChartPoint alloc] initWithX:[NSNumber numberWithInt:i]
                                                              y:[NSNumber numberWithDouble:u * s]]];
                [points addObject:[[ChartPoint alloc] initWithX:[NSNumber numberWithInt:i + 1]
                                                              y:[NSNumber numberWithDouble:v * s]]];
                break;
            }
        } while (true);
    }
    
    return points;
}

+ (NSMutableArray *)generateRandomData:(NSInteger)count {
    NSMutableArray *points = [[NSMutableArray alloc] init];
    
    for (int i = 0; i < count ; i++) {
        double random = arc4random()%10000 + 100;
        [points addObject:[[ChartPoint alloc] initWithX:[NSNumber numberWithInt:i*10]
        y:[NSNumber numberWithDouble:random]]];
    }
    
    return points;
}

@end

File: Swift/FlexChart101/FlexChart101/ChartData.swift
//
//  ChartData.swift
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

import UIKit

class ChartData: NSObject {
    var name: String
    var sales: Double
    var expenses: Double
    var downloads: Double
    
    init(name: String, sales: Double, expenses: Double, downloads: Double) {
        self.name = name
        self.sales = sales
        self.expenses = expenses
        self.downloads = downloads
    }
    
    class func demoData() -> NSMutableArray {
        return[
            ChartData(name: "US", sales: Double(arc4random()%10001), expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Germany", sales: Double(arc4random()%10001), expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "UK", sales: Double(arc4random()%10001), expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Japan", sales: Double(arc4random()%10001), expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Italy", sales: Double(arc4random()%10001), expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Greece", sales: Double(arc4random()%10001), expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001))
        ]
    }
    
    class func annotationData() -> NSMutableArray {
        return[
            ChartData(name: "US", sales: 1500, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Germany", sales: 4000, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "UK", sales: 3000, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Japan", sales: 6000, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Italy", sales: 3500, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Greece", sales: 8500, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "France", sales: 2300, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Spain", sales: 6500, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Ireland", sales: 4500, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001)),
            ChartData(name: "Poland", sales: 9500, expenses: Double(arc4random()%5001), downloads: Double(arc4random()%20001))
        ]
    }
}

class ChartPoint : NSObject {
    var x: Double
    var y: Double
    
    init(x: Double, y: Double) {
        self.x = x
        self.y = y
    }
    
    class func generateRandomPoints(count: Int) -> NSMutableArray {
        var points = NSMutableArray()
        var innerCount = count
        
        if innerCount % 2 == 1 {
            innerCount++
        }
        
        for var i = 0; i < innerCount / 2; i++ {
            repeat
            {
                let random: Double = Double(rand()) / Double(RAND_MAX)
                let u: Double = 2 * random - 1
                let v: Double = 2 * random - 1
                
                var s: Double = u * u + v * v
                
                if s < 1
                {
                    s = sqrt(-2 * log(s) / s)
                    points.addObject(ChartPoint(x: Double(i), y: (u * s)))
                    points.addObject(ChartPoint(x: Double(i + 1), y: (v * s)))
                    break
                }
            } while true
        }
        
        return points
    }
    
    class func generateRandomData(count: Int) -> NSMutableArray {
        let points = NSMutableArray()

        
        for var i = 0; i < count; i++ {
            var random :Double = (Double)(arc4random()%10000) + 100.0
            points.addObject(ChartPoint.init(x: (Double)(i*10), y: random))
        }
        
        return points
    }
}
